test/e2e/catalog_e2e_test.go

```diff
@@ -1340,6 +1343,61 @@ var _ = Describe("Starting CatalogSource e2e tests", func() {
 			}
 		}
 	})
+
+	FWhen("A CatalogSource is created with an operator that has a CSV with missing metadata.ApiVersion", func() {
+
+		var (
+			magicCatalog      MagicCatalog
+			catalogSourceName string
+			subscription      *operatorsv1alpha1.Subscription
+			c                 client.Client
+		)
+
+		BeforeEach(func() {
+			c = ctx.Ctx().Client()
+
+		})
+
+		//AfterEach(func() {
+		//	TeardownNamespace(ns.GetName())
+		//})
+		It("properly delpoys a catalogSource using the resources", func ()  {
+
+			provider, err := NewFileBasedFiledBasedCatalogProvider(filepath.Join(testdataDir, badCSVDir, "bad-csv.yaml"))
+			Expect(err).To(BeNil())
+			fmt.Printf("Provider: %v\n", provider)
+
+			catalogSourceName = genName("cat-bad-csv")
+			magicCatalog = NewMagicCatalog(c, ns.GetName(), catalogSourceName, provider)
+			fmt.Printf("magicCatalog: %v\n", magicCatalog)
+			Expect(magicCatalog.DeployCatalog(context.Background())).To(BeNil())
+			fmt.Printf("namespace: %s\n\n", ns.GetName())
+			
+		})
```

Comment:
I mentioned this in slack but this looks like a potential scoping issue:

- We're building up and deploying a catalog that contains a bundle with a bad CSV in the test generated namespace
- Because we're wrapping this logic in an It spec vs. a BeforeEach container, when the `"A Subscription is created catalogSource built with the malformed CSV"` test case is run, the namespace that contains this catalog has already been deleted, and a new namespace has been created.

I think we only need to perform some light refactoring of this current setup to get everything working as intended:

> Note: I jotted this down inline so YMMV w.r.t compilation errors

```go
		BeforeEach(func() {
			c = ctx.Ctx().Client()

			provider, err := NewFileBasedFiledBasedCatalogProvider(filepath.Join(testdataDir, badCSVDir, "bad-csv.yaml"))
			Expect(err).To(BeNil())

			catalogSourceName = genName("cat-bad-csv")
			magicCatalog = NewMagicCatalog(c, ns.GetName(), catalogSourceName, provider)
			Expect(magicCatalog.DeployCatalog(context.Background())).To(BeNil())
		})

		When("A Subscription is created catalogSource built with the malformed CSV", func() {
			It("fails with a ResolutionFailed error condition, and a message that highlights the missing field in the CSV", func() {
				subscription = &operatorsv1alpha1.Subscription{
					ObjectMeta: metav1.ObjectMeta{
						Name:      fmt.Sprintf("%s-sub", catalogSourceName),
						Namespace: ns.GetName(),
					},
					Spec: &operatorsv1alpha1.SubscriptionSpec{
						CatalogSource:          catalogSourceName,
						CatalogSourceNamespace: ns.GetName(),
						Channel:                "stable",
						Package:                "packageA",
					},
				}
				Expect(c.Create(context.Background(), subscription)).To(BeNil())

				_, err := fetchSubscription(crc, subscription.GetNamespace(), subscription.GetName(), subscriptionHasInstallPlanChecker)
				Expect(err).Should(BeNil())
			})
		})
```

Reply:
That fixed it. I am forever in your debt.